### PR TITLE
Fix mailer extension runtime package

### LINKED ISSUE #3950: Improve consistency of the Mailer extension packages
Right now, we end up with the config reference doc being named `quarkus-mailer-impl.adoc` due to these inconsistencies.

We could work around it in the config reference doc generation but having consistency is better.

### PR DESCRIPTION
**Pull request by gsmet**
For consistency with the naming rules.

Fixes #3950

### CODE REVIEW

File: extensions/mailer/deployment/src/main/java/io/quarkus/mailer/deployment/MailerProcessor.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import io.quarkus.deployment.builditem.LaunchModeBuildItem;
 import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
 import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
-import io.quarkus.mailer.impl.*;
+import io.quarkus.mailer.runtime.*;
```
Comment (machi1990) [inline extensions/mailer/deployment/src/main/java/io/quarkus/mailer/deployment/MailerProcessor.java]: Does not look related but I think  we can avoid using start imports here :-)

**Review by machi1990 — APPROVED** [review]
LGTM - except for that one minor comment.

File: extensions/mailer/deployment/src/main/java/io/quarkus/mailer/deployment/MailerProcessor.java (hunk 1 above)
Comment (gsmet) [reply extensions/mailer/deployment/src/main/java/io/quarkus/mailer/deployment/MailerProcessor.java]: Yeah, it was preexisting but I pushed an update to fix it.

**Review by geoand — APPROVED** [review]

**Pull request merged**